app/services/update_subscription_address.rb

```diff
@@ -0,0 +1,28 @@
+class UpdateSubscriptionAddress
+
+  def update_address(subscription, subscription_params)
+    new_ship_address = subscription_params[:ship_address_attributes]
+    new_bill_address = subscription_params[:bill_address_attributes]
+
+    if (subscription.ship_address != new_ship_address) && (subscription.bill_address == new_bill_address)
+      update_ship_address(subscription, new_ship_address)
+    elsif (subscription.bill_address != new_bill_address) && (subscription.ship_address == new_ship_address)
+      update_bill_address(subscription, new_bill_address)
+    elsif (subscription.ship_address != new_ship_address) && (subscription.bill_address != new_bill_address)
+      update_ship_and_bill_address(subscription, new_ship_address, new_bill_address)
+    end
+  end
+
+  def update_ship_address(subscription, new_ship_address)
+    subscription.ship_address.update_columns(new_ship_address)
+  end
+
+  def update_bill_address(subscription, new_bill_address)
+    subscription.bill_address.update_columns(new_bill_address)
+  end
```

Comment:
The address on an order should be `readonly`, but I agree with @aamyot here, the `SubscriptionAddress` can/should be editable.